public/index.ts

```diff
@@ -991,12 +993,12 @@ function createTestVolume() {
   };
 }
 
-function createLoader(data: TestDataSpec): IVolumeLoader {
+async function createLoader(data: TestDataSpec): Promise<IVolumeLoader> {
   switch (data.type) {
     case "opencell":
       return new OpenCellLoader();
     case "omezarr":
-      return new OMEZarrLoader(data.url, volumeCache);
+      return await OMEZarrLoader.createLoader(data.url, 0, volumeCache);
```

Comment:
Thinking about how to keep this a bit more polymorphic across all loaders, there are a couple of things you could do.
1. make a factory method in VolumeLoaderUtils that figures out file type from url and creates the loader. (more or less replicating this function) - and then always create loaders through that function, even in w3cv
2. make all loaders constructors private and make them all have a static createLoader.
3. let the constructors stay trivial and add an async init call to be called right after construction
Some mix and match of those would keep this code more uniform in appearance.  I slightly favor #1 if it works in w3cv and here.  You might have to pass in more info than url in order for it to detect properly.

Reply:
I'll give 1 a try, it's also my favorite